### ISSUE TITLE
Support for Source map

### ISSUE THREAD

**Issue by SeabassNg** [gabrysbiz/lesscss-extended-compiler#3]
Would be nice to have a source map to point where in the less file the code is from

**Comment by agabrys** [comment]
This library is not responsible for compiling sources (and also generating source maps) ;-) It only wraps native compiler (see [lesscss-compiler](https://github.com/gabrysbiz/lesscss-compiler)) and extends it (files cache and different protocols support). I think that you can generate source map by using [compiler options](http://lesscss.org/usage/#command-line-usage-source-map-output-filename) - I will check this.   If this is available, then I will try to prepare nice API for this :-)

**Comment by agabrys** [comment]
Unfortunately I found a bug related with source map when using rhino - [less.js #2044](https://github.com/less/less.js/issues/2044).

Fortunately I solve this problem ;-) You need to add compiler options:

```
--source-map=<source-file> --source-map-output-map-file=<source-map-full-path>
```

example:

```
File source = new File("/root/src/main/less/basic.less");
LessCompiler compiler = new LessCompilerImpl();
String[] arguments = {
    "--source-map=basic.less",
    "--source-map-output-map-file=/root/target/basic.css.map"
};
String code = compiler.compile(source, new CompilerOptions(Arrays.asList(arguments)));
FileUtils.write(new File("/root/target/basic.css"), code);
```

Probably I will rewrite [lesscss-compiler](https://github.com/gabrysbiz/lesscss-compiler) to be more source map friendly and next apply those changes to this library.

**Comment by agabrys** [comment]
New API will be implemented in:
[lesscss-compiler](https://github.com/gabrysbiz/lesscss-compiler) 2.0
[lesscss-extended-compiler](https://github.com/gabrysbiz/lesscss-extended-compiler) 3.0
[lesscss-maven-plugin](https://github.com/gabrysbiz/lesscss-maven-plugin) 2.0

**Comment by agabrys** [comment]
Blocked by [lesscss-compiler #3](https://github.com/gabrysbiz/lesscss-compiler/issues/3).

**Comment by agabrys** [comment]
Today I started working on Less Compiler 2.0.0. It will handle Source Maps. This library will be deprecated after release of Less Compiler 2.0.0, so I marked this bug as "Won't fix".

See:
* [gabrysbiz/lesscss-compiler#milestone 2.0.0](/gabrysbiz/lesscss-compiler/milestone/2)
* gabrysbiz/lesscss-compiler#3
